tests/phpunit/Writer/MSSQLEntrypointTest.php

```diff
@@ -328,6 +392,63 @@ public function testExceptionLogging(): void
         $this->assertContains('"class":"Keboola\\\\DbWriter\\\\Application"', $process->getErrorOutput());
     }
 
+    public function testRetry(): void
+    {
+        $testDataSet = 'runIncremental';
+        $config = $this->initConfig($testDataSet);
+
+        // Modify config
+        unset($config['parameters']['tables'][0]);
+        unset($config['parameters']['tables'][2]);
+
+        $orgConfig = $config;
+        $table = $orgConfig['parameters']['tables'][1];
+
+        $config['parameters']['db']['user'] = self::NO_PERM_USER_LOGIN;
+        $config['parameters']['db']['#password'] = self::NO_PERM_USER_PASSWORD;
+        $this->initInputFiles($testDataSet, $config);
+
+        // Create table
+        $writer = $this->getWriter($orgConfig['parameters']);
+        $writer->create($table);
+
+        // Insert to table is denied
+        $this->conn->exec(sprintf(
+            'GRANT CONTROL ON DATABASE::%s TO %s',
+            'test',
+            self::NO_PERM_USER_LOGIN
+        ));
+        $this->conn->exec(sprintf(
+            'GRANT CONTROL ON SCHEMA::%s TO %s',
+            'dbo',
+            self::NO_PERM_USER_LOGIN
+        ));
+        $this->conn->exec(sprintf(
+            'DENY INSERT ON OBJECT::%s.simple TO %s',
+            'dbo',
+            self::NO_PERM_USER_LOGIN
+        ));
+
+        // Run app
+        $process = $this->runApp();
+        $output = $process->getOutput();
+        $errorOutput = $process->getErrorOutput();
+        $expectedError =
+            'The INSERT permission was denied ' .
+            'on the object \'simple\', database \'test\', schema \'dbo\'.. ';
+
+        // Check retry in output
+        $this->assertEquals(1, $process->getExitCode(), $process->getOutput());
+        $this->assertStringContainsString($expectedError . 'Retrying... [1x]', $output);
```

Comment:
Pridany test na `retry`, chyba v `INSERT INTO` sa simuluje tak, ze dany pouzivatel ma vsetky prava okrem zapisania do danej tabulky ... teda sa tam deje retry, a nakoniec to spadne.